app/app.go

```diff
@@ -186,18 +200,62 @@ func NewInitApp(
 		app.SupplyKeeper,
 		app.subspaces[clp.ModuleName])
 
+	skipUpgradeHeights := make(map[int64]bool)
+	skipUpgradeHeights[0] = true
+	app.UpgradeKeeper = upgrade.NewKeeper(skipUpgradeHeights, keys[upgrade.StoreKey], app.cdc)
+	app.UpgradeKeeper.SetUpgradeHandler("testupgrade", func(ctx sdk.Context, plan upgrade.Plan) {
+		ethAsset := clp.NewAsset("Ethereum", "ETH", "ceth")
+		pool, err := clp.NewPool(ethAsset, sdk.NewUint(100), sdk.NewUint(100), sdk.NewUint(10))
+		if err != nil {
+			return
+		}
+		err = app.clpKeeper.SetPool(ctx, pool)
+		if err != nil {
+			return
+		}
+		clp.InitGenesis(ctx, app.clpKeeper, clp.DefaultGenesisState())
+	})
+	app.SetStoreLoader(bam.StoreLoaderWithUpgrade(&types.StoreUpgrades{
+		Added: []string{clp.ModuleName},
+	}))
```

Comment:
@Jedi2002 - this looks like some test logic, no?

Reply:
THis PR is not ment to be merged , This piece of code is used to make sure the steps in the document 
https://docs.google.com/document/d/1AqWfSBK8m8XFi23aQ5j3SLhO7AcFWFKG0SWRjUDT3EE/edit , works as expected .

Adding the upgrade module in itself does not need any code, We would just need to write snippents like this when we want to do live upgrades  , the snipets would define the migration ( old binary -> new binary )  .

Lastly this snippet is pretty simple ,real upgrades would require much more complicated migrationss ,and every upgrade would be unique to the the old and new state .

Reply:
@Jedi2002 Gotcha. So far it all looks good to me. One thing I'd be keen on us doing is using the auto-download feature:

https://github.com/cosmos/cosmos-sdk/tree/master/cosmovisor#auto-download

thoughts??

Reply:
Yeah , I tried using that , But I was not really sure if that would be reliable  . I am thinking along the lines , that validators might have different configurations in terms of firewals ,os security etc , and the auto download feature might get blocked due to that .

I feel publishing documents which a validator can follow might be better , becuase in that case the we can pinpoint the issue easier , debugging the system security params for a validator might not be something we would get into . 

Would love to discuss more on this though @intl-man